src/runtime/sitemap/urlset/filter.ts

```diff
@@ -2,9 +2,12 @@ import { parseURL } from 'ufo'
 import { createRouter, toRouteMatcher } from 'radix3'
 import type { ModuleRuntimeConfig, ResolvedSitemapUrl, SitemapDefinition } from '../../types'
 
+interface RegexObjectType {
+  regex: string
+}
 interface CreateFilterOptions {
-  include?: (string | RegExp)[]
-  exclude?: (string | RegExp)[]
+  include?: (string | RegexObjectType)[]
```

Comment:
we want to provide a straightforward DX to end users, so they should just be able to provide the RegExp, the RegexObjectType is just an internal transform for the module to deal with a data type that can't be serialized out of the box

So ideally we support `(string | RegExp | RegexObjectType)[]`

this requires transforming the filter rules, so within module.ts we can just normalise the rules within the setup

```ts
config.include = (config.include || []).map(r => r instanceof RegExp ? { regex: r.toString() } : r)
```

Reply:
Totally right, ill do the changes